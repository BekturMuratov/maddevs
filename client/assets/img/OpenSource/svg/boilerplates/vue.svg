<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>