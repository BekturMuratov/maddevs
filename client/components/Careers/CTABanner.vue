<template>
  <div class="careers-cta-banner_container container">
    <UIBanner
      title="Can’t see your next role?"
      text="If you don’t see a suitable vacancy, but you are sure we’re meant to be, drop us a line."
      :image="require(`@/assets/img/Careers/Daria.png`)"
      image-alt-text="Daria"
      man-name="Daria Utesheva"
      man-position="Head of HR Departament"
    >
      <a
        href="https://t.me/DianaHRDigital"
        target="_blank"
      >
        <UIButton class="careers-cta-banner__button">
          <img
            :data-src="require(`@/assets/img/Careers/svg/telegram--white.svg`)"
            width="20"
            height="17"
            class="img_lazy"
          >
          &nbsp;
          Telegram us
        </UIButton>
      </a>
      <ul class="careers-cta-banner__links">
        <li>
          <a
            href="#"
            target="_blank"
          >
            Headhunter
          </a>
        </li>
        <li>
          <a
            href="#"
            target="_blank"
          >
            Habr.com
          </a>
        </li>
        <li>
          <a
            href="#"
            target="_blank"
          >
            Linkedin
          </a>
        </li>
      </ul>
    </UIBanner>
  </div>
</template>

<script>
import UIBanner from '@/components/shared/UIBanner'
import UIButton from '@/components/shared/UIButton'

export default {
  name: 'CTABanner',
  components: {
    UIBanner,
    UIButton,
  },
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

.careers-cta-banner {
  &_container {
    padding-bottom: 100px;
  }
  &__button {
    font-weight: normal;
    padding: 17px 31px;
    font-size: 18px;
    line-height: 18px;
    letter-spacing: -0.02em;
    img {
      transform: translateY(2.5px);
    }
    @media screen and (max-width: 900px) {
      font-size: 16px;
    }
  }
  &__links {
    margin-top: 24.5px;
    display: flex;
    li {
      margin-right: 11.5px;
      a {
        letter-spacing: -0.013em;
        font-size: 14px;
        line-height: 19px;
        color: $text-color--black-oil;
        text-decoration: underline;
      }
      &:last-child {
        margin-right: 0;
      }
    }
  }
}
</style>
