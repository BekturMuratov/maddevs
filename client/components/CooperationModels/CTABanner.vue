<template>
  <div class="cta-banner_container container">
    <UIBanner
      title="Build your product <br /> with Mad Devs"
      text="Together we build solutions that bring digital innovation."
      :image="require(`@/assets/img/CooperationModels/png/tamara.png`)"
      image-alt-text="Tamara"
      man-name="Tamara Mun"
      man-position="Delivery Manager"
    >
      <UIButton @click="$refs.modalContactMe.show()">
        Estimate your project
      </UIButton>
    </UIBanner>
    <Modal ref="modalContactMe">
      <ModalContactMe @success="$refs.modalContactMe.close()" />
    </Modal>
  </div>
</template>

<script>
import UIBanner from '@/components/shared/UIBanner'
import UIButton from '@/components/shared/UIButton'
import Modal from '@/components/core/Modal'

export default {
  name: 'CTABanner',
  components: {
    UIBanner,
    UIButton,
    Modal,
    ModalContactMe: () => import('@/components/core/modals/ModalContactMe'),
  },
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

.cta-banner {
  &_container {
    padding-bottom: 119px;
    @media screen and (max-width: 768px) {
      padding-bottom: 44px;
    }
  }

  /deep/ .ui-button {
    width: 233px;
    height: 52px;
    font-weight: 400;
    font-size: 18px;
    letter-spacing: -0.02em;
    @media screen and (max-width: 900px) {
      font-size: 16px;
    }
  }
}
</style>
