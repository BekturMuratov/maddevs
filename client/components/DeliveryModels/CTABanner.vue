<template>
  <div class="cta-banner_container container">
    <UIBanner
      title="Build your product <br /> with Mad Devs"
      text="Together we build solutions that bring digital innovation."
      :image="require(`@/assets/img/DeliveryModels/png/Tamara.png`)"
      image-alt-text="Tamara"
      man-name="Tamara Mun"
      man-position="Delivery Manager"
    >
      <UIButton @click="$refs.modalContactMe.show()">
        Estimate your project
      </UIButton>
    </UIBanner>
    <ModalContactMe ref="modalContactMe" />
  </div>
</template>

<script>
import UIBanner from '@/components/shared/UIBanner'
import UIButton from '@/components/shared/UIButton'

export default {
  name: 'CTABanner',
  components: {
    UIBanner,
    UIButton,
    ModalContactMe: () => import('@/components/core/modals/ModalContactMe'),
  },
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

.cta-banner_container {
  padding-bottom: 119px;
  @media screen and (max-width: 768px) {
    padding-bottom: 44px;
  }

  /deep/ .cta-banner {
    min-height: 543px;
    @media screen and (max-width: 1180px) {
      min-height: 490px;
    }
    @media screen and (max-width: 992px) {
      min-height: 450px;
    }
  }

  /deep/ .ui-button {
    width: 233px;
    height: 52px;
    font-weight: 400;
    font-size: 18px;
    letter-spacing: -0.02em;
    @media screen and (max-width: 900px) {
      font-size: 16px;
    }
  }

}
</style>
