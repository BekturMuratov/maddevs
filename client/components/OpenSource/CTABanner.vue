<template>
  <div>cta</div>
</template>

<script>
export default {
  name: 'CTABanner',
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

</style>
