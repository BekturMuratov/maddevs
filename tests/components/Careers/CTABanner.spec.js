import CTABanner from '@/components/Careers/CTABanner'
import { render } from '@testing-library/vue'

const stubs = ['UIBanner', 'UIButton']

const mocks = {
  $getMediaFromS3: () => 'img.png',
  $t: () => 'translated',
}

const directives = {
  'lazy-load': () => {},
}

describe('CTABanner component', () => {
  it('should render correctly', () => {
    const { container } = render(CTABanner, {
      stubs,
      mocks,
      computed: {
        image: () => 'image.png',
      },
      directives,
    })

    expect(container).toMatchSnapshot()
  })
})
