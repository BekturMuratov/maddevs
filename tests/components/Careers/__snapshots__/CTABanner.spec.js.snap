// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTABanner component should render correctly 1`] = `
<div>
  <div
    class="careers-cta-banner_container container"
    data-aos="fade-up"
  >
    <uibanner-stub
      image="[object Object]"
      imagealttext="Daria"
      manname="Daria Utesheva"
      manposition="Head of HR Departament"
      text="If you don’t see a suitable vacancy, but are sure that we’re meant to be, drop us a line."
      title="Can’t find your <br /> next role?"
    >
      <a
        href="https://t.me/DianaHRDigital"
        target="_blank"
      >
        <uibutton-stub>
          <img
            class="img_lazy"
            data-src="[object Object]"
            height="17"
            width="20"
          />
          
         
        Telegram us
      
        </uibutton-stub>
      </a>
       
      <ul
        class="careers-cta-banner__links"
      >
        <li>
          <a
            href="https://career.habr.com/companies/maddevs/cp/vacancies/"
            target="_blank"
          >
            
          Headhunter
        
          </a>
        </li>
         
        <li>
          <a
            href="https://novosibirsk.hh.ru/employer/2861993/"
            target="_blank"
          >
            
          Habr.com
        
          </a>
        </li>
         
        <li>
          <a
            href="https://www.linkedin.com/company/mad-devs/"
            target="_blank"
          >
            
          Linkedin
        
          </a>
        </li>
      </ul>
    </uibanner-stub>
  </div>
</div>
`;
